pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1320,6 +1320,13 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                                                             new SubscriptionNotFoundException(
                                                                     "Subscription does not exist"));
                                         }
+                                        if (((AbstractTopic) topic).isConsumersExceededOnTopic()) {
```

Comment:
Could we move this before to the auto subscription creation check?

Reply:
Good suggestion. Fixed